nemo_asr/calc_rtf_chunk.py

```diff
@@ -0,0 +1,224 @@
+chunk_len = 30
+total_audio_len = 600 
+MODEL = args.model
+batch_size = args.batch_size
+nbatches = args.nbatches
+warmup_batches = args.warmup_batches
+decoding_type = args.decoding_type
+model_stride = 8  # 8 for fastconformer and citrinet, 4 for conformer
+chunk_batch_size = 24  # number of chunks to run in parallel
+DEVICE=torch.device(args.gpu)
+logging.info(f'MODEL: {MODEL}')
+
+def get_samples(audio_file, total_audio_len, target_sr=16000):
+    with sf.SoundFile(audio_file, 'r') as f:
+        dtype = 'int16'
+        sample_rate = f.samplerate
+        samples = f.read(dtype=dtype)
+        if sample_rate != target_sr:
+            samples = librosa.core.resample(samples, orig_sr=sample_rate, target_sr=target_sr)
+        samples = samples.astype('float32') / 32768
+        samples = samples.transpose()
+        sample_length = samples.shape[0]
+        if sample_length > total_audio_len * target_sr:
+            logging.info(f'resizing audio sample from {sample_length / target_sr} to maxlen of {total_audio_len}')
+            sample_length = int(total_audio_len * target_sr)
+            samples = samples[:sample_length]
+            logging.info(f'new sample lengh: {samples.shape[0]}')
+        else:
+            pad_length = int(total_audio_len * target_sr) - sample_length
+            logging.info(f'padding audio sample from {sample_length / target_sr} to maxlen of {total_audio_len}')
+            samples = np.pad(samples, (0, pad_length), 'constant', constant_values=(0, 0))
+            sample_length = int(total_audio_len * target_sr)
+            
+        return samples, sample_length
+
+
+def extract_preprocessor(model, device):
+    cfg = copy.deepcopy(model._cfg)
+    OmegaConf.set_struct(cfg.preprocessor, False)
+    cfg.preprocessor.dither = 0.0
+    cfg.preprocessor.pad_to = 0
+    preprocessor = model.from_config_dict(cfg.preprocessor)
+    return preprocessor.to(device)
+
+def setup_aed_decoding(asr_model):
+    decoding_cfg = MultiTaskDecodingConfig()
+    decoding_cfg.strategy = "beam"
+    decoding_cfg.beam.beam_size = 1
+    asr_model.change_decoding_strategy(decoding_cfg)
+
+def setup_rnnt_decoding(asr_model):
+    decoding_cfg = RNNTDecodingConfig()
+    decoding_cfg.strategy = "greedy_batch"
+    if hasattr(asr_model, 'cur_decoder'):
+        asr_model.change_decoding_strategy(decoding_cfg, decoder_type="rnnt")
+    else:
+        asr_model.change_decoding_strategy(decoding_cfg)
+
+def setup_ctc_decoding(asr_model):
+    decoding_cfg = CTCDecodingConfig()
+    decoding_cfg.strategy = "greedy"
+    if hasattr(asr_model, 'cur_decoder'):
+        asr_model.change_decoding_strategy(decoding_cfg, decoder_type="ctc")
+    else:
+        asr_model.change_decoding_strategy(decoding_cfg)
+
+def setup_rnnt_chunk_infer(frame_asr, audio_input):
+    frame_reader = AudioFeatureIterator(audio_input, frame_asr.frame_len, frame_asr.raw_preprocessor, frame_asr.asr_model.device)
+    frame_asr.set_frame_reader(frame_reader)
```

Comment:
minor ... this common frame asr can be moved to new func for reduced redundancy,